src/test/java/com/ministore/api/integration/CreateUserTest.java

```diff
@@ -51,4 +51,39 @@ void mustBeAbleToCreateAUser() throws Exception
             .andExpect(jsonPath("$.id").value(response.id))
             .andExpect(jsonPath("$.username").value(response.username));
     }
+
+    @Test
+    void mustReturnNotFoundWhenUserNotFound() throws Exception {
+        this.mockMvc.perform(get("/user/abc"))
+            .andExpect(status().isNotFound());
+    }
+
+    @Test
+    void mustReturnBadRequestWhenCreateUserDoesntReceivePayload() throws Exception {
+        this.mockMvc.perform(
+            post("/user")
+            .contentType(MediaType.APPLICATION_JSON)
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    void mustReturnBadRequestWhenUsernameIsInvalid() throws Exception {
+        this.mockMvc.perform(
+            post("/user")
+            .contentType(MediaType.APPLICATION_JSON)
+            .content("{\"username\":\"\"}")
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    void mustReturnBadRequestWhenPayloadDoesntHaveUsername() throws Exception {
+        this.mockMvc.perform(
+            post("/user")
+            .contentType(MediaType.APPLICATION_JSON)
+            .content("{}")
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isBadRequest());
+    }
```

Comment:
Nice tests!

I'd like to ask you to think about two new cases, and add tests accordingly to cement the behavior:

* What should happen when the same username is sent more than once? i.e. what does the system do if we make the `POST` request with body `"{\"username\":\"ricardotulio\"}"` twice?
* What should happen when the request body includes more fields than the username? i.e.: what does the system do if we receive a body like this: `"{\"username\":\"some.username\", password: \"some password\"}"`?

Reply:
Following our conversation today, can you add a test that checks the second case?